doc/source/contributing/syntax_guide.md

```diff
@@ -0,0 +1,201 @@
+(syntax_guide)=
+# Syntax Guide
+
+## Adding targets
+Adding custom targets or anchors to the headings is really helpful for cross-referencing. They allow us to link to the heading using a simple syntax.
+They are defined using this syntax:
+
+::::{tab-set}
+
+:::{tab-item} rST
+:sync: rst
+
+```
+.. _mytarget:
+```
+:::
+
+:::{tab-item} MyST (Markdown)
+:sync: myst
+
+```
+(mytarget)=
+```
+:::
+
+::::
+
+They are referred using this syntax:
+
+::::{tab-set}
+
+:::{tab-item} rST
+:sync: rst
+
+```
+:ref:`mytarget`
+```
+:::
+
+:::{tab-item} MyST (Markdown)
+:sync: myst
+
+```
+{ref}`mytarget`
+```
+:::
+
+::::
+
+For adding anchors in `.ipynb` files, Markdown syntax will be used in the markdown cells of `.ipynb` files. Read more about it [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role).
+
+## Backticks for highlighting code keywords
+
+For highlighting inline code keywords or file names, backticks are used. In Markdown single backticks are used while in rST double backticks are used. For example, for highlighting the file name `conf.py`, we will use this syntax:
+
+````{tabbed} rST
+```
+``conf.py``
+```
+````
+````{tabbed} MyST (Markdown)
+```
+`conf.py`
+```
+````
+
+## Table of content tree
+
+You might have noticed that ArviZ docs maintain a hierarchy to keep the documentation organized. A table of content tree is added to all the main pages to make the website easy to navigate. Adding the table of content tree provides the list of all pages in the left sidebar. It also enables navigation to the **Previous** and **Next** pages in the footer.
+
+Follow this syntax for adding the table of content:
+
+rST:
+<pre>
```

Comment:
This should be a triple backtick block. This pre makes it look similar but it is not understood as code by sphinx. Therefore it is not modified by sphinx_copybutton for example. You can see on the preview how the copy button doesn't appear: https://arviz--1903.org.readthedocs.build/en/1903/contributing/syntax_guide.html#table-of-content-tree